StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Embedded/EmbeddedPaymentElement+Internal.swift

```diff
@@ -394,35 +395,20 @@ extension EmbeddedPaymentElement: EmbeddedFormViewControllerDelegate {
     }
 
     func embeddedFormViewControllerDidCancel(_ embeddedFormViewController: EmbeddedFormViewController) {
-        // If the formViewController was populated with a previous payment option don't reset
-        if embeddedFormViewController.previousPaymentOption == nil {
-            let lastSelection = embeddedPaymentMethodsView.previousSelectedRowButton?.type
-            var currentlySelectedType = embeddedPaymentMethodsView.selectedRowButton?.type
-            if let lastSelection, lastSelection != currentlySelectedType {
-                // Go back to the previous selection if there was one
-                embeddedPaymentMethodsView.resetSelectionToLastSelection()
-            } else {
-                // If there wasn't a previous selection, deselect if the selection isn't valid.
-                let isCurrentSelectionValid = embeddedFormViewController.selectedPaymentOption != nil
-                if !isCurrentSelectionValid {
-                    embeddedPaymentMethodsView.resetSelection()
-                }
-            }
+        // Go back to the previous selection if there was one
```

Comment:
I ticketed up the work to restore the previous form when you cancel https://jira.corp.stripe.com/browse/MOBILESDK-3361